File: src/main/java/no/nav/fo/forenkletdeploy/teams/TeamOppfolging.java
package no.nav.fo.forenkletdeploy.teams;

import no.nav.fo.forenkletdeploy.domain.ApplicationConfig;
import org.slf4j.Logger;
import org.slf4j.LoggerFactory;

import java.util.ArrayList;
import java.util.Arrays;
import java.util.List;
import java.util.Map;

import static java.util.stream.Collectors.toList;
import static no.nav.json.JsonUtils.fromJson;
import static no.nav.sbl.rest.RestUtils.withClient;

public class TeamOppfolging implements Team {
    private Logger logger = LoggerFactory.getLogger(TeamOppfolging.class.getName());
    public static final String APPLICATION_CONFIG_URL = "http://stash.devillo.no/projects/OPP/repos/team-oppfolging/raw/applikasjonsportefolje/config.json";

    private List<ApplicationConfig> applicationConfigs = new ArrayList<>();

    @Override
    public String getId() {
        return "teamoppfolging";
    }

    @Override
    public String getDisplayName() {
        return "Team Oppfølging";
    }

    @Override
    public String getJenkinsFolder() {
        return "teamoppfolging";
    }

    @Override
    public List<ApplicationConfig> getApplicationConfigs() {
        return this.applicationConfigs;
    }

    @Override
    public void hentApplicationConfigs() {
        try {
            String json = withClient(c -> c.target(APPLICATION_CONFIG_URL).request().get(String.class));
            Map<String, Map<String, String>> map = fromJson(json, Map.class);
            this.applicationConfigs = map.entrySet().stream()
                    .map(entry -> ApplicationConfig.builder()
                            .name(entry.getKey())
                            .gitUrl(entry.getValue().get("gitUrl"))
                            .build()
                    )
                    .collect(toList());
        } catch (Throwable e) {
            logger.error("Feil ved henting av application-configs for team-oppfølging");
        }
    }
}

File: src/main/kotlin/no/nav/fo/forenkletdeploy/teams/TeamOppfolging.kt
package no.nav.fo.forenkletdeploy.teams

import no.nav.fo.forenkletdeploy.ApplicationConfig
import no.nav.fo.forenkletdeploy.util.Utils.fromJson
import no.nav.fo.forenkletdeploy.util.Utils.withClient
import org.slf4j.LoggerFactory

import java.util.ArrayList

class TeamOppfolging : Team {
    private val logger = LoggerFactory.getLogger(TeamOppfolging::class.java.name)
    val APPLICATION_CONFIG_URL = "http://stash.devillo.no/projects/OPP/repos/team-oppfolging/raw/applikasjonsportefolje/config.json"

    override var applicationConfigs: List<ApplicationConfig> = ArrayList()
    override val id = "teamoppfolging"
    override val displayName = "Team Oppfølging"
    override val jenkinsFolder = "teamoppfolging"

    override fun hentApplicationConfigs() {
        try {
            val json = withClient(APPLICATION_CONFIG_URL).request().get(String::class.java)
            val map = fromJson(json, OppfolgingAppConfig::class.java)
            this.applicationConfigs = map.entries
                    .map { ApplicationConfig( name = it.key, gitUrl = it.value.gitUrl ) }
        } catch (e: Throwable) {
            logger.error("Feil ved henting av application-configs for team-oppfølging")
        }

    }
}

data class OppfolgingConfig(
        val gitUrl: String = ""
)

class OppfolgingAppConfig: HashMap<String, OppfolgingConfig>()